### PR TITLE
Upgrades node 22 and pnpm

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9968

## Description of Changes
- Upgrades to node 22
- Upgrades pnpm to 9.14.2
- Fixes typing issues
- Refactors Outbox typings

## Test Plan
- Unit tested the `FIXME()` call.
- CA (click around) tested on local and frack:
  - TODO page
  - 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: libs/core/src/integration/outbox.schema.ts [hunk 1]
```diff
@@ -10,161 +9,42 @@ const BaseOutboxProperties = z.object({
...
+
+export const Outbox = z.union(
+  Object.entries(outboxEvents).map(([event_name, event_payload]) =>
+    z
+      .object({
+        event_name: z.literal(event_name as keyof typeof outboxEvents),
+        event_payload,
+      })
+      .merge(BaseOutboxProperties),
+  ) as unknown as readonly [z.ZodTypeAny, z.ZodTypeAny, ...z.ZodTypeAny[]],
```
Comment (timolegros) [inline libs/core/src/integration/outbox.schema.ts:49]: While this is more concise, it completely removes the types on the Outbox schema. Do we really want to do that?

Comment (timolegros) [reply libs/core/src/integration/outbox.schema.ts:49]: I just saw your comment regarding the size of the type. I guess this is fine as long as we make sure to primarily use the `emitEvent` util.

**Review by timolegros — CHANGES_REQUESTED** [review]
Update the version in `.nvmrc`

File: libs/core/src/integration/outbox.schema.ts (hunk 1 above)
Comment (Rotorsoft) [reply libs/core/src/integration/outbox.schema.ts:49]: Yeah, I think we only need it to validate. I didn't find anything using these types.

**Review by kurtisassad — APPROVED** [review]

**Review by timolegros — DISMISSED** [review]
Getting this error when I run `pnpm start`. I have updated to Node22 using `nvm` and my `pnpm` version is also updated to `9.14.2`.
```
[21:09:52.187] ERROR (port.ts): Unhandled Rejection 
    err: {
      "type": "Error",
      "message": "The module '/home/timolegros/Projects/commonwealth/node_modules/.pnpm/better-sqlite3@11.1.2/node_modules/better-sqlite3/build/Release/better_sqlite3.node'\nwas compiled against a different Node.js version using\nNODE_MODULE_VERSION 115. This version of Node.js requires\nNODE_MODULE_VERSION 127. Please try re-compiling or re-installing\nthe module (for instance, using `npm rebuild` or `npm install`).",
      "stack":
          Error: The module '/home/timolegros/Projects/commonwealth/node_modules/.pnpm/better-sqlite3@11.1.2/node_modules/better-sqlite3/build/Release/better_sqlite3.node'
          was compiled against a different Node.js version using
          NODE_MODULE_VERSION 115. This version of Node.js requires
          NODE_MODULE_VERSION 127. Please try re-compiling or re-installing
          the module (for instance, using `npm rebuild` or `npm install`).
              at Object..node (node:internal/modules/cjs/loader:1715:18)
              at Module.load (node:internal/modules/cjs/loader:1318:32)
              at Function._load (node:internal/modules/cjs/loader:1128:12)
              at TracingChannel.traceSync (node:diagnostics_channel:315:14)
              at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
              at Module.require (node:internal/modules/cjs/loader:1340:12)
              at require (node:internal/modules/helpers:141:16)
              at bindings (/home/timolegros/Projects/commonwealth/node_modules/.pnpm/bindings@1.5.0/node_modules/bindings/bindings.js:112:48)
              at new Database (/home/timolegros/Projects/commonwealth/node_modules/.pnpm/better-sqlite3@11.1.2/node_modules/better-sqlite3/lib/database.js:48:64)
              at new ModelDB (file:///home/timolegros/Projects/commonwealth/node_modules/.pnpm/@canvas-js+modeldb-sqlite@0.12.1/node_modules/@canvas-js/modeldb-sqlite/lib/ModelDB.js:28:19)
      "code": "ERR_DLOPEN_FAILED"
    }
```

Fixed it with `pnpm` rebuild. As part of this PR you should provide instructions in `all-eng` about upgrading Node + Pnpm and then possibly running the rebuild.

**Review by timolegros — CHANGES_REQUESTED** [review]
I think the Node version in the Docker builds needs to be updated as well. See `Dockerfile.datadog`.

**Review by timolegros — APPROVED** [review]

**Pull request merged**